homeassistant/components/peco/sensor.py

```diff
@@ -0,0 +1,117 @@
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    websession = hass.data[DOMAIN][config_entry.entry_id]["websession"]
+    conf: MappingProxyType[str, Any] = config_entry.data
+
+    async def async_update_data():
+        """Fetch data from API."""
+        try:
+            async with async_timeout.timeout(10):
+                if conf["county"] == "TOTAL":
+                    return await api.get_outage_totals(websession)
+                else:
+                    return await api.get_outage_count(conf["county"], websession)
+        except InvalidCountyError as err:
+            raise ConfigEntryAuthFailed from err
+        except HttpError as err:
+            raise UpdateFailed from err
+        except BadJSONError as err:
+            raise UpdateFailed from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="PECO Outage Count",
+        update_method=async_update_data,
+        update_interval=timedelta(minutes=SCAN_INTERVAL),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    _LOGGER.info("Setting up sensor platform")
+    county = conf["county"]
+    _LOGGER.info("County: %s", county)
+    async_add_entities(
+        [PecoOutageCounterSensorEntity(hass, config_entry.title, county, coordinator)],
+        True,
+    )
+    return
+
+
+class PecoOutageCounterSensorEntity(CoordinatorEntity, SensorEntity):
+    """PECO outage counter sensor."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_icon: str = "mdi:power-plug-off"
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        county: str,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.hass = hass
+        self._county = county
+        self._name = name
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def native_value(self) -> int:
+        """Return the value of the sensor."""
+        return self.coordinator.data["customers_out"]
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        if self.coordinator.data["percent_customers_out"] < 5:
+            percent_customers_out = "Less than 5%"
+        else:
+            percent_customers_out = (
+                str(self.coordinator.data["percent_customers_out"]) + "%"
+            )
+        return {
+            "percent_customers_out": percent_customers_out,
+            "outage_count": self.coordinator.data["outage_count"],
+            "customers_served": self.coordinator.data["customers_served"],
+        }
+
+    @property
+    def unique_id(self) -> str:
```

Comment:
This can also be defined in `__init__()` with an `self._attr_unique_id`.
When you have multiple entities you'd have to add a key (like `f"{country}_"customers_out"` or preferably define the key in the SensorEntityDescriptions `key` attribute and use it here)